mathjax3-ts/output/chtml/Wrappers/menclose.ts

```diff
@@ -0,0 +1,951 @@
+import {BBox} from '../BBox.js';
+import {MmlMenclose} from '../../../core/MmlTree/MmlNodes/menclose.js';
+import {MmlNode, AbstractMmlNode, AttributeList} from '../../../core/MmlTree/MmlNode.js';
+import {OptionList} from '../../../util/Options.js';
+import {StyleList, StyleData} from '../CssStyles.js';
+import {split} from '../../../util/string.js';
+
+/*****************************************************************/
+
+const ARROWX = 4, ARROWDX = 1, ARROWY = 2;  // default relative arrowhead values
+
+const THICKNESS = .067;  // default rule thickness
+const PADDING = .2;      // default padding
+
+const SOLID = THICKNESS + 'em solid';  // a solid border
+
+/*
+ * The functions used for notation definitions
+ */
+export type Renderer<N, T, D> = (node: CHTMLmenclose<N, T, D>, child: N) => void;
+export type BBoxExtender<N, T, D> = (node: CHTMLmenclose<N, T, D>) => number[];
+export type BBoxBorder<N, T, D> = (node: CHTMLmenclose<N, T, D>) => number[];
+export type Initializer<N, T, D> = (node: CHTMLmenclose<N, T, D>) => void;
+
+/*
+ * The definition of a notation
+ */
+export type NotationDef<N, T, D> = {
+    renderer: Renderer<N, T, D>;        // renders the HTML for the notation
+    bbox: BBoxExtender<N, T, D>;        // gives the offsets to the child bounidng box: [top, right, bottom, left]
+    border?: BBoxBorder<N, T, D>;       // gives the amount of the bbox offset that is due to borders on the child
+    renderChild?: boolean;              // true if the notation is used to render the child directly (e.g., radical)
+    init?: Initializer<N, T, D>;        // function to be called during wrapper construction
+    remove?: string;                    // list of notations that are suppressed by this one
+};
+
+/*
+ * For defining notation maps
+ */
+export type NotationDefPair = [string, NotationDef<any, any, any>];
+export type NotationDefList = Map<string, NotationDef<any, any, any>>;
+
+/*
+ * The list of notations for an menclose element
+ */
+export type NotationList<N, T, D> = {[notation: string]: NotationDef<N, T, D>};
+
+
+/*****************************************************************/
+
+/*
+ * The names and indices of sides for borders, padding, etc.
+ */
+export const sideIndex = {top: 0, right: 1, bottom: 2, left: 3};
+export type Side = keyof typeof sideIndex;
+
+/*
+ * the spacing to leave for an arrowhead
+ */
+export const arrowHead = (node: CHTMLmenclose<any, any, any>) => {
+    return Math.max(node.padding, node.thickness * (node.arrowhead.x + node.arrowhead.dx + 1));
+};
+
+/*
+ * Service routines used for notations
+ */
+export const Notation = {
+
+    /*
+     * Common BBox and Border functions
+     */
+    fullBBox: ((node) => new Array(4).fill(node.thickness + node.padding)) as BBoxExtender<any, any, any>,
+    fullPadding: ((node) => new Array(4).fill(node.padding)) as BBoxExtender<any, any, any>,
+    fullBorder: ((node) => new Array(4).fill(node.thickness)) as BBoxBorder<any, any, any>,
```

Comment:
Should all these really be exported? Or are they not just private to this module?